docs: Add flatten_separator parameter to flatten processor (#12879)
opensearch-project/documentation-website: _data-prepper/pipelines/configuration/processors/flatten.md

`remove_processed_fields` | No | Boolean | When `true`, the processor removes all processed fields from the source. Default is `false`.
`remove_list_indices` | No | Boolean | When `true`, the processor converts the fields from the source map into lists and puts the lists into the target field. Default is `false`.
`flatten_when` | No | String | A [conditional expression]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/expression-syntax/), such as `/some-key == "test"'`, that determines whether the `flatten` processor will be run on the event. Default is `null`, which means that all events will be processed unless otherwise stated.

> `flatten_separator` | No | String | The character used to join nested key names into flattened keys. Must be a single character. Default is `"."`.

Reviewer: Should we add something saying "since 2.17" here?

Author: Is that something we want to do for our documentation? I do not think we have done that at all

Reviewer: We probably should add some sort of tag, but as this doesn't yet exist, this is fine.

> `flatten_separator` | No | String | The character used to join nested key names into flattened keys. Must be a single character. Default is `"."`.

Reviewer 2: ```suggestion
`flatten_separator` | No | String | The character used to join nested key names into flattened keys. Must be a single character. Default is `.`.
```

  "list1[].list2[].value": ["value1","value2"]
}
```

### Custom separator

Use the `flatten_separator` option to specify a custom character for joining nested key names. The default separator is `.`, but you can use any single character, such as `_`, as shown in the following example:

```yaml
...
  processor:
    - flatten:
        source: "log"
        target: ""
        flatten_separator: "_"
        remove_processed_fields: true
...
```
{% include copy.html %}

> For example, when the input event contains the following nested objects:

Reviewer 2: ```suggestion
Consider an input event that contains the following nested objects:
```
